madprops/serializers.py

```diff
@@ -52,7 +54,12 @@ def to_representation(self, value):
         return result_dict
 
     def to_internal_value(self, data):
-        data_list = [{name: value} for (name, value) in data.items()]
+        try:
+            data_list = [{name: value} for (name, value) in data.items()]
+        except AttributeError:
+            raise ValidationError(
+                _('`%(data)s` is not a key value mapping') % {'data': data})
```

Comment:
Updated message.

Marked as translation because it gives flexibility to translate error messages with no major drawback. This repo is public and it's not inconceivable we'd want to return error messages in different languages.